lucille-plugins/lucille-weaviate/src/main/java/com/kmwllc/lucille/weaviate/stage/DropIndexedDocuments.java

```diff
@@ -0,0 +1,279 @@
+
+    // initialize fields to get from Weaviate
+    Field[] fields = new Field[]{
+        Field.builder()
+            .name(idDestinationName)
+            .build(),
+        Field.builder()
+            .name("_additional")
+            .fields(
+                Field.builder()
+                    .name("id")
+                    .build()
+            ).build()
+    };
+
+    // initialize a request
+    Get query = client
+        .graphQL()
+        .get()
+        .withClassName(weaviateClassName)
+        .withFields(fields)
+        .withLimit(batchSize);
+
+    // if lastID is not null, add it to the request for pagination
+    if (lastID != null) {
+      query.withAfter(lastID);
+    }
+
+    // run query
+    Result<GraphQLResponse> response = query.run();
+
+    if (response.hasErrors()) {
+      throw new StageException("Error getting ids from Weaviate: " + response.getError().toString());
+    }
+
+    JsonArray array = GSON.toJsonTree(response.getResult().getData())
+        .getAsJsonObject()
+        .get("Get")
+        .getAsJsonObject()
+        .get(weaviateClassName)
+        .getAsJsonArray();
+
+    List<String> parsedIDs = new ArrayList<>();
+    for (int i = 0; i < array.size(); i++) {
+      JsonObject obj = array.get(i).getAsJsonObject();
+      JsonObject additional = obj.get("_additional").getAsJsonObject();
+
+      parsedIDs.add(obj.get("id_original").getAsString());
+
+      if (i == array.size() - 1) {
+        lastID = additional.get("id").getAsString();
+      }
+    }
+
+    return new Pair<>(parsedIDs, lastID);
+  }
+
+  private int getCount() throws StageException {
+
+    Field fields = Field.builder()
+        .name("meta")
+        .fields(
+            Field.builder()
+                .name("count")
+                .build()
+        ).build();
+
+    Result<GraphQLResponse> response = client
+        .graphQL()
+        .aggregate()
+        .withClassName(weaviateClassName)
+        .withFields(fields)
+        .run();
+
+    if (response.hasErrors()) {
+      throw new StageException("Error getting count from Weaviate: " + response.getError().toString());
+    }
+
+    JsonArray array = GSON.toJsonTree(response.getResult().getData())
+        .getAsJsonObject()
+        .get("Aggregate")
+        .getAsJsonObject()
+        .get(weaviateClassName)
+        .getAsJsonArray();
+
+    if (array.size() != 1) {
+      throw new StageException("Expected 1 result from Weaviate, got " + array.size());
+    }
+
+    return array
+        .get(0)
+        .getAsJsonObject()
+        .get("meta")
+        .getAsJsonObject()
+        .get("count")
+        .getAsInt();
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) {
+    String id = doc.getId();
+
+    if (ids.contains(id)) {
+
+      // set drop document to true
+      doc.setDropped(true);
+
+      // remove from set
+      ids.remove(id);
```

Comment:
what if two docs with the same id are generated by the connector?
suggest not removing id from set -- leave it there in case the same id comes up again